src/org/traccar/BaseEventHandler.java

```diff
@@ -12,24 +14,32 @@ public boolean isLastPosition() {
         return isLastPosition;
     }
 
+    private Device device;
```

Comment:
I do not argue, but may be I misunderstand something with OOP.
`anylizePosition` is called just after setting `device` in the same object instance in the same function. `device` can be different?